core/src/test/java/kafka/server/share/SharePartitionTest.java

```diff
@@ -3554,6 +3555,907 @@ public void testLsoMovementThenAcquisitionLockTimeoutThenAcknowledgeBatchLastOff
         assertNull(sharePartition.cachedState().get(5L).batchAcquisitionLockTimeoutTask());
     }
 
+    @Test
+    public void testWriteShareGroupStateWithNullResponse() {
+        Persister persister = Mockito.mock(Persister.class);
+        mockPersisterReadStateMethod(persister);
+        SharePartition sharePartition = SharePartitionBuilder.builder().withPersister(persister).build();
+
+        Mockito.when(persister.writeState(Mockito.any())).thenReturn(CompletableFuture.completedFuture(null));
+        assertThrows(IllegalStateException.class, () -> sharePartition.isWriteShareGroupStateSuccessful(Mockito.anyList()));
```

Comment:
Agreed with this change @chia7712, but that will only work if we agree to my above comment [here](https://github.com/apache/kafka/pull/16530#discussion_r1668662377), otherwise it will simply return a true/false value here instead of an exception

Reply:
> but that will only work if we agree to my above comment https://github.com/apache/kafka/pull/16530#discussion_r1668662377

agree :smile: